src/utils/resolveHOC.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ *
+ */
+
+import recast from 'recast';
+import isReactCreateClassCall from './isReactCreateClassCall';
+
+var {
+  types: {
+    NodePath,
+    namedTypes: types,
+  },
+} = recast;
+
+/**
+ * If the path is a call expression, it recursively resolves to the
+ * rightmost argument, stopping if it finds a React.createClass call expression
```

Comment:
That's interesting that Relay doesn't follow the convention I've seen for every other HOC, which is to take only one argument (the component) or have the component be the rightmost argument. Recompose follows this convention.

But as discussed in https://github.com/reactjs/react-docgen/issues/80 you can easily turn any HOC into a HOC that takes a component as its only or rightmost argument.